### PR TITLE
Actionable user can cancel

### PR DESCRIPTION
**Pull request by pkarman**
trello: https://trello.com/c/j4c8Lxfp/318-ability-of-tier-2-approvers-to-cancel-a-request

### CODE REVIEW

File: spec/policies/proposal_policy_spec.rb [hunk 1]
```diff
@@ -162,9 +162,21 @@
...
 
-    it "doesn't allow an approver to cancel it" do
-      proposal = create(:proposal, :with_parallel_approvers, :with_observers)
-      expect(ProposalPolicy).not_to permit(proposal.approvers[0], proposal)
+    it "allows an actionable approver to cancel it" do
+      proposal = create(:proposal, :with_serial_approvers, :with_observers)
+      expect(ProposalPolicy).to permit(proposal.approvers[0], proposal)
+    end
+
+    it "allows a delegate of an actionable approver to cancel it" do
```
Comment (jessieay) [inline spec/policies/proposal_policy_spec.rb]: we have a transient `delegate` attr on the `proposal` factory so you could shorten this:

```ruby
delegate = create(:user)
proposal = create(:proposal, delegate: delegate)
expect(ProposalPolicy).to permit(delegate, proposal)
```

File: spec/policies/proposal_policy_spec.rb [hunk 2]
```diff
@@ -162,9 +162,21 @@
       expect(ProposalPolicy).not_to permit(proposal.requester, proposal)
     end
 
-    it "doesn't allow an approver to cancel it" do
-      proposal = create(:proposal, :with_parallel_approvers, :with_observers)
-      expect(ProposalPolicy).not_to permit(proposal.approvers[0], proposal)
+    it "allows an actionable approver to cancel it" do
+      proposal = create(:proposal, :with_serial_approvers, :with_observers)
```
Comment (jessieay) [inline spec/policies/proposal_policy_spec.rb]: do we need observers for this spec?

File: spec/policies/proposal_policy_spec.rb [hunk 3]
```diff
@@ -162,9 +162,21 @@
...
+
+    it "allows a delegate of an actionable approver to cancel it" do
+      proposal = create(:proposal, :with_serial_approvers, :with_observers)
+      delegate = create(:user)
+      proposal.approvers[0].add_delegate(delegate)
+      expect(ProposalPolicy).to permit(delegate, proposal)
+    end
+
+    it "does not allow pending approver to cancel it" do
+      proposal = create(:proposal, :with_serial_approvers, :with_observers)
```
Comment (jessieay) [inline spec/policies/proposal_policy_spec.rb]: do we need observers?

File: spec/features/cancel_spec.rb [hunk 4]
```diff
@@ -48,6 +48,26 @@
     expect(current_path).to eq("/proposals/#{proposal.id}/cancel_form")
   end
 
+  context "step completers" do
+    it "allows actionable step completer to cancel" do
```
Comment (jessieay) [inline spec/features/cancel_spec.rb]: should we have a spec like this for a delegate too?

**Comment by pkarman** [comment]
@jessieay test added, observers zapped.

**Comment by jessieay** [comment]
:rainbow:

**Pull request merged**